src/wallet/rpcdump.cpp

```diff
@@ -1413,17 +1413,21 @@ UniValue importmulti(const JSONRPCRequest& mainRequest)
         }
     }
     if (fRescan && fRunScan && requests.size()) {
-        int64_t scannedTime = pwallet->RescanFromTime(nLowestTimestamp, reserver, true /* update */);
+        CWallet::ScanResult result = pwallet->RescanFromTime(nLowestTimestamp, reserver, true /* update */);
         {
             auto locked_chain = pwallet->chain().lock();
             LOCK(pwallet->cs_wallet);
             pwallet->ReacceptWalletTransactions();
         }
 
-        if (pwallet->IsAbortingRescan()) {
+        if (result.status == CWallet::ScanResult::USER_ABORT) {
             throw JSONRPCError(RPC_MISC_ERROR, "Rescan aborted by user.");
-        }
-        if (scannedTime > nLowestTimestamp) {
+        } else if (result.status == CWallet::ScanResult::FAILURE) {
+            int64_t scannedTime = TIMESTAMP_MIN;
+            if (!pwallet->chain().findBlock(result.last_failed_block, nullptr /* block */, nullptr /* time */, &scannedTime)) {
+                throw std::logic_error("ScanForWalletTransactions returned invalid block hash");
+            }
+            scannedTime += TIMESTAMP_WINDOW + 1;
```

Comment:
This differs from the original code, where `scannedTime` was set to the last block time + `TIMESTAMP_WINDOW + 1`. Won't this result in missed scans below, as `scannedTime` will be low?

Reply:
I don't see the difference in behavior - see the original location for comparison: https://github.com/bitcoin/bitcoin/pull/14942/commits/f059899fe3ca88204d9dda4f83c4f8c4548743ea#diff-b2bb174788c7409b671c46ccc86034bdL1607-L1611

Could you call out what you're referring to any more explicitly, in terms of what has changed?

Reply:
No, sorry, you're right. I missed the passing of `scannedTIme` on L#1427. It looks identical